File: Python/Leetcode/202_happy_number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        b=[]
        for i in range(7):
            if n==1:
                return True
            a=list(str(n))
            n=sum(pow(int(i),2) for i in a)
            b.append(n)
        if len(set(b))!=len(b):
            return False

File: C++/Leetcode/202_happy_number.cpp
class Solution {
int squareAdder(int n)
{
    int sum = 0;
    while (n != 0)
    {
        int lastDigit = n % 10;
        sum += pow(lastDigit, 2);
        n = n / 10;
    }
    return sum;
}

public:
bool isHappy(int n)
{
    unordered_set<int> s;
    s.insert(n);
    if (n == 1)
        return true;
    while (n != 1)
    {
        n = squareAdder(n);
        if (n == 1)
            return true;
        else if (s.find(n) != s.end())
        {
            return false;
        }
        else
        {
            s.insert(n);
        }
    }
    return false;
}
};
